tools/osquery-agent-options/README.md

```diff
@@ -5,3 +5,17 @@ This directory contains a script (a Go command) that generates the struct needed
 It prints the resulting Go code to stdout (the `osqueryOptions` and the `osqueryCommandLineFlags` structs), you can just copy it and insert it in the proper location in the source code to replace the existing struct (in `server/fleet/agent_options.go`).
 
 Note that the latest version of osquery should be installed for this tool to work properly (`osqueryd` and `osqueryi` must be in your $PATH).
+
+## OS-specific flags
+
+Some osquery flags are OS-specific and will not show up either with `osqueryd --help` or with the `osqueryi` query, depending on the OS you're running those on. In the code (in `server/fleet/agent_options.go`), those OS-specific flags are defined in the `OsqueryCommandLineFlags{Linux,MacOS,Windows}` structs, and the `osquery-agent-options` tool will automatically ignore from its generated struct any flag already defined as part of one of the OS-specific structs.
+
+It can be hard to even know what OS-specific flags exist, because of the fact they don't show up in `osqueryd --help` or the `osqueryi` query when not running that specific OS, and the fact that not all flags are documented in [the osquery docs](https://osquery.readthedocs.io/en/stable/). To help with this, the following bash command can be executed assuming you have the latest osquery repository cloned locally:
+
+```
+# ag is the Silver Searcher, a grep alternative, but it should work with grep too, maybe
+# with some small adjustments to the flags.
+$ ag --nofilename -o 'FLAGS_[a-z0-9_]+' | sort | uniq | cut -d _ --complement -f 1
+```
+
+This finds all flags defined in the osquery codebase (assuming all flags are built the same way). It is then possible to run a diff of this list with the list from the `osqueryi` query (e.g. `osqueryi --list 'select name from osquery_flags;'`), and the missing ones are possibly/likely OS-specific. It's not an automatable task, as some judgement and manual code inspection may be necessary (some flags may be just in a test file, there may be some false-positives like `FLAGS_start` and `FLAGS_end` that are only sentinel values, the code line may be commented-out, etc.), but at least it gives a list of potential such flags.
```

Comment:
I think it could but the thing is that we don't control this codebase and each version could introduce a new subtle way that breaks our automation, basically I think that even if we automated it, we'd have to manually check with the next version to make sure that automation still makes sense, that we don't rely on something silently broken?

> I'm not sure if it's worth the effort to investigate.

From what I've seen doing this PR, I don't think it's worth the (considerable, I think) effort to automate that part. What we _could_ do, which would be low-effort, is to keep the output of those `FLAGS_....` found in the last version, so we'd just have to compare the output of the new version with the old one to see if there are any new hidden/os-specific flags. What do you think? I could add that file to this PR.

Reply:
all of that makes sense to me 👍 

>  is to keep the output of those FLAGS_.... found in the last version, so we'd just have to compare the output of the new version with the old one to see if there are any new hidden/os-specific flags. What do you think? I could add that file to this PR.

this makes sense, and could be the thing used for the CI check we're discussing in a concurrent thread :D

Reply:
> could be the thing used for the CI check 

True! Allright, I'll add this file to the PR and document its use.